lib/optimization/tableid.go

```diff
@@ -0,0 +1,63 @@
+package optimization
+
+import (
+	"fmt"
+
+	"github.com/artie-labs/transfer/lib/config/constants"
+	"github.com/artie-labs/transfer/lib/sql"
+	"github.com/artie-labs/transfer/lib/stringutil"
+)
+
+type TableIdentifier struct {
+	database string
+	schema   string
+	name     string
+}
+
+func NewTableIdentifier(database, schema, name string) TableIdentifier {
+	return TableIdentifier{database: database, schema: schema, name: name}
+}
+
+func (t TableIdentifier) Name(uppercaseEscNames bool, args *sql.NameArgs) string {
+	return sql.EscapeName(t.name, uppercaseEscNames, args)
+}
+
+type FqNameOpts struct {
+	BigQueryProjectID   string
+	MsSQLSchemaOverride string
+}
+
+func (t TableIdentifier) FqName(kind constants.DestinationKind, escape bool, uppercaseEscNames bool, opts FqNameOpts) string {
```

Comment:
The only changes here should be:
- `t.TopicConfig.Database` -> `t.database`
-  `t.TopicConfig.Schema` -> `t.schema`